compiler/resolution/wrappers.cpp

```diff
@@ -71,16 +84,26 @@ static void       reorderActuals(FnSymbol*                fn,
 static void       coerceActuals(FnSymbol* fn,
                                 CallInfo& info);
 
-static bool       isPromotionRequired(FnSymbol* fn, CallInfo& info);
+static bool       isPromotionRequired(FnSymbol* fn, CallInfo& info,
+                                std::vector<ArgSymbol*>& actualIdxToFormal);
 
 static FnSymbol*  promotionWrap(FnSymbol* fn,
                                 CallInfo& info,
+                                std::vector<ArgSymbol*>& actualIdxToFormal,
                                 bool      buildFastFollowerChecks);
 
-static FnSymbol*  buildEmptyWrapper(FnSymbol* fn, CallInfo& info);
+static FnSymbol*  buildEmptyWrapper(FnSymbol* fn);
 
 static ArgSymbol* copyFormalForWrapper(ArgSymbol* formal);
 
+typedef struct DefaultExprFnEntry_s {
+  FnSymbol* defaultExprFn;
+  std::vector<std::pair<ArgSymbol*,ArgSymbol*> > usedFormals;
+} DefaultExprFnEntry;
+
+typedef std::map<ArgSymbol*, DefaultExprFnEntry> formalToDefaultExprEntryMap;
+formalToDefaultExprEntryMap formalToDefaultExprEntry;
+
```

Comment:
While there, could you clarify the comment below about 'actualIdxToFormals' ?
The comment says it contains the actuals, however that cannot be true
because an actual is not necessarily an ArgSymbol.

Understanding actualIdxToFormals would help understanding wrapAndCleanUpActuals()
as well as addDefaultsAndReorder().